src/misc/ethernet_slirp.cpp

```diff
@@ -0,0 +1,267 @@
+/* *  Copyright (C) 2020  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+
+#if C_SLIRP
+
+#include "ethernet_slirp.h"
+#include <time.h>
+#include "dosbox.h"
+
+#ifdef WIN32
+#include <ws2tcpip.h>
+#else
+#include <arpa/inet.h>
+#endif
+
+ssize_t slirp_send_packet(const void *buf, size_t len, void *opaque)
+{
+	SlirpEthernetConnection* conn = (SlirpEthernetConnection*)opaque;
+	conn->Receive_Packet((uint8_t*)buf, len);
+	return len;
+}
+
```

Comment:
The function name `slirp_send_packet` implies it sends a packet, so makes sense that the inbound buffer is **const** (because we're not writing to it).

But then inside the function, we actually `Receive_Packet`, and cast-away the const? `(uint8_t*)buf`? Presumably because `Receive_Packet()` would write to the buffer.

Either the calls are mixed up or their names are very confusing(?)

Reply:
> The callback slirp uses is slirp_send_packet, for sending a packet back to us. So we receive the packet and send it to the NE2000. The names are confusing, but it's a slirp thing.

Thanks for that description! Suggest adding it to the code for future maintainers.

Reagrding the `const *buf` but then cast-away-the-const, can this be fixed? ie: just take in a `uint8_t *buf` or `char *buf` from the get-go? Otherwise it's pretty ugly to cast-away-const.

If it's absolutely impossible to solve, suggest using `const_cast< ... >`.